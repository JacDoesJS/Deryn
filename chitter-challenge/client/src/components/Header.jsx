import logo from "./images/logo.png";
import { useNavigate } from 'react-router-dom';
import { useEffect, useState } from 'react';
import Login from "./Login.jsx";

const Header = () => {
    const navigate = useNavigate();
    const handleClickPeep = () => navigate('/addPeep');
    const handleClickAllPeeps = () => navigate('/allPeeps');
    const [user, setLoginUser] = useState({});


    const navigateToAllPeeps = (e) => {
        e.preventDefault();
        navigate('/allPeeps');
    };

    const navigateToAddPeep = (e) => {
        e.preventDefault()
        navigate('/addPeep');
    };

    const navigateToLogin = (e) => {
        e.preventDefault()
        navigate('/login');
    };

    return (

        <div className="head">

            <div className="logo">
                <img src={logo} alt="logo" width="100%" />
            </div>
            <div className="buttons">
                <form action="./" method="post">
                    <input id="loginout" onClick={navigateToLogin} className="button" type="submit" value="login/out" name="loginout"></input>
                    <input id="peep" onClick={navigateToAddPeep} className="button" type="submit" value="PEEP" name="peep"></input>
                    <input id="all-peeps" onClick={navigateToAllPeeps} className="button" type="submit" value="all peeps" name="all-peeps"></input>
                </form>

            </div>

        </div>

    );
}

export default Header;