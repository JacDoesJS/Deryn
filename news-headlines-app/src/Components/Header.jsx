import banner from "./images/banner3.webp"
import guardian from "./images/guardian.png"

const Header = () => {
    return (
        <header className="App-header">

            <nav className="head">
                <div className="container-fluid">
                    <p id="dfx">DFX News for you</p>
                </div>
            </nav>
            <div className="banner">
                <img src={banner} alt="logo" />
            </div>
            <h2 className="topTitle">Breaking News <img src={guardian} alt="Guardian logo" width="200"></img></h2>

        </header>
    );
}

export default Header;