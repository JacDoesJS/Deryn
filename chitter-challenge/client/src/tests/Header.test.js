import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../Components/Header';

test(`Header matches snapshot`, () => {
    // const headerComponent = render(<Header />);

    // expect(headerComponent).toMatchSnapshot();
    expect(render(<MemoryRouter><Header /></MemoryRouter>)).toMatchSnapshot();
});