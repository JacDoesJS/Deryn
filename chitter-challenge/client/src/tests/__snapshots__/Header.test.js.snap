// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="head"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="logo.png"
            width="100%"
          />
        </div>
        <div
          class="buttons"
        >
          <form
            action="./"
            method="post"
          >
            <input
              class="button"
              id="account"
              name="account"
              type="submit"
              value="login/out"
            />
            <input
              class="button"
              id="peep"
              name="peep"
              type="submit"
              value="PEEP"
            />
            <input
              class="button"
              id="all-peeps"
              name="all-peeps"
              type="submit"
              value="all peeps"
            />
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="head"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="logo.png"
          width="100%"
        />
      </div>
      <div
        class="buttons"
      >
        <form
          action="./"
          method="post"
        >
          <input
            class="button"
            id="account"
            name="account"
            type="submit"
            value="login/out"
          />
          <input
            class="button"
            id="peep"
            name="peep"
            type="submit"
            value="PEEP"
          />
          <input
            class="button"
            id="all-peeps"
            name="all-peeps"
            type="submit"
            value="all peeps"
          />
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
